sbnana/CAFAna/Core/EnsembleSpectrum.h

```diff
@@ -14,12 +14,14 @@ class TGraphAsymmErrors;
 namespace ana
 {
   class EnsembleRatio;
+  class EnsembleFluxTimesNuclei;
   class FitMultiverse;
 
   class EnsembleSpectrum : public beta::IValueEnsembleSink
   {
   public:
     friend class EnsembleRatio;
+    friend class EnsembleFluxTimesNuclei;
```

Comment:
This seems like it will be a problem, since eventually the Ensemble classes will be in cafanacore, while FluxTimesNuclei will remain an experiment-specific thing.

Reply:
EnsembleFluxTimesNuclei inherits, so I don't think this should be necessary.

Reply:
So I tried without and it would not let me access the protected constructor of the base class from within the derived class. For `Spectrum` the equivalent constructor is public which would solve this issue.

Reply:
I just did a little experiment with this and was surprised this is the case. But what it looks like you can do is access the protected base class constructor from a derived class constructor. That will work for you, right? Make a (protected) constructor for `EnsembleFluxTimesNuclei` that mostly forwards to that `EnsembleSpectrum` constructor.

One day we might be forced to make the eigen constructor public, but I'm not very happy about exposing the internal details of how the universes are laid out in the array.

Reply:
Ah, I did miss that. Actually, it doesn't seem too unreasonable to return EnsembleFluxTimesNuclei from here. The thing is still a flux, and in principle the PDG code is still reasonable. Either EnsembleSpectrum / EnsembleFluxTimesNuclei already gives you an EnsembleRatio automatically, or you can immediately assign the return value from MakeTotalFlux. As far as I can tell the only odd thing about the resulting object is that passing the BinDensity argument to ToTH1 will give the wrong axis label, which I think we can live with.

Reply:
Pushed a fix that hopefully clears this up?